### PR TITLE
Add initial benchmark framework

### PR DESCRIPTION
**Pull request by t4ccer**
This replicates interface of `tasty-bench` basically one to one but you give it `ClosedTerm`s instead

### CODE REVIEW

**Review by kozross — APPROVED** [review]

File: plutarch-testlib/bench/Bench.hs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          [ bench
+              "non-optimized"
+              (plam (\m -> pmaybeToMaybeData #$ pmaybeDataToMaybe # m) # pconstant @(PMaybeData PInteger) (Just 42))
+          , bcompare
+              "$2 == \"pmaybeToMaybeData . pmaybeDataToMaybe\" && $3 == \"non-optimized\""
+              $ benchWithConfig
+                "optimized"
+                Optimizing
+                (plam (\m -> pmaybeToMaybeData #$ pmaybeDataToMaybe # m) # pconstant @(PMaybeData PInteger) (Just 42))
+          ]
```
Comment (SeungheonOh) [inline plutarch-testlib/bench/Bench.hs:24]: please also add examples using baseline here.

Comment (t4ccer) [reply plutarch-testlib/bench/Bench.hs:24]: It already supports that, baseline is done via CLI flags. You first generate csv file with `cabal run bench -- --csv baseline.csv` and then with no modifications to bench code you can do `cabal run bench -- --baseline baseline.csv`

Comment (t4ccer) [reply plutarch-testlib/bench/Bench.hs:24]: There's `cabal run bench -- --help` also

Comment (kozross) [reply plutarch-testlib/bench/Bench.hs:24]: Yeah, those are directly taken from how `tasty-bench` works.

**Review by SeungheonOh — APPROVED** [review]
LGTM. Tasty's pattern stuff is little annoying but it works well besides slight annoyances

**Comment by t4ccer** [comment]
```
$ cabal run test:tests --ghc-options=-Wwarn -- --quickcheck-replay="(SMGen 946588892800887496 14704601251472569737,51)" --pattern 'V3.Ledger Laws PTxInfo.can parse toData of original'
Tests
  PlutarchLedgerApi
    Laws
      V3
        Ledger Laws PTxInfo
          can parse toData of original: OK (14.77s)
            +++ OK, passed 62 tests.
```

**Review by SeungheonOh — APPROVED** [review]
Oopsie with generator. Changes looks good.

**Pull request merged**